### PR TITLE
[FIX] Select Rows: None on output when no data

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
**Rank** widget crashes because receives empty `Table` and empty `Domain` which is created by **Select Rows**.

##### Description of changes
**Select Rows** commits `None`.

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/386054859/

File: Orange/widgets/data/owselectrows.py [hunk 1]
```diff
@@ -530,8 +530,11 @@ def commit(self):
                 matching_output = remover(matching_output)
                 non_matching_output = remover(non_matching_output)
 
-        self.Outputs.matching_data.send(matching_output)
-        self.Outputs.unmatched_data.send(non_matching_output)
+        def output_filter(output):
+            return output if output is not None and len(output) else None
```
Comment (janezd) [inline Orange/widgets/data/owselectrows.py]: I here agree with @astaric's usual comment about my code. If-else expressions in Python are mostly unreadable. I'd prefer

```
if output is not None and len(output):
    return output
```

You don't even need `return None` here, but you may add it to make it explicit.

Comment (janezd) [reply Orange/widgets/data/owselectrows.py]: Oh, besides, I don't like the function at all. I'd prefer having

```
if not len(matching_output):
    matching_output = None
```

and the same for `non_matching_output`.

The same amount of code, but way more explicit.

**Comment by janezd** [comment]
I made the change. Another option is to allow empty tables, but not empty domains. In this case we don't call `remover` if `not len(data)`.

**Pull request merged**